activestorage/test/models/blob_test.rb

```diff
@@ -157,6 +157,17 @@ class ActiveStorage::BlobTest < ActiveSupport::TestCase
     end
   end
 
+  test "open when checksum is not available, skip integrity check" do
+    create_blob(data: "Hello, world!").tap do |blob|
+      blob.update_attribute(:checksum, nil)
+      assert blob.checksum.blank?
+
+      assert_nothing_raised do
+        blob.open { |file| assert file }
+      end
```

Comment:
1. Assert the expected outcome rather than erecting guard rails. If there's a regression, the integrity exception would cause the test to fail.
2. The assertion within the block would silently no-op if Blob#open skipped opening unverifiable files.

Consider e.g.
```ruby
create_blob(data: "unverified").tap do |blob|
  blob.update_attribute :checksum, nil

  assert_equal "unverified", blob.open(&:read)
end
```